hail/src/main/scala/is/hail/expr/ir/MatrixIR.scala

```diff
@@ -2051,6 +2051,19 @@ case class MatrixExplodeRows(child: MatrixIR, path: IndexedSeq[String]) extends
   }
 }
 
+case class MatrixRepartition(child: MatrixIR, n: Int, shuffle: Boolean) extends MatrixIR {
+  val typ: MatrixType = child.typ
+
+  def children: IndexedSeq[BaseIR] = FastIndexedSeq(child)
+
+  def copy(newChildren: IndexedSeq[BaseIR]): MatrixRepartition = {
+    val IndexedSeq(newChild: MatrixIR) = newChildren
+    MatrixRepartition(newChild, n, shuffle)
+  }
+
+  override def columnCount: Option[Int] = child.columnCount
```

Comment:
Where's the `execute` method?

Reply:
it's not there! This is executed using table lowering:

https://github.com/hail-is/hail/pull/4838/files#diff-05fc91db2b386f679f151ae746b209acR160